extensions/docker-desktop/README.md

```diff
@@ -62,3 +61,28 @@ follow these steps to have it recognized as a CLI plugin under `docker`:
 mkdir -p ~/.docker/cli-plugins
 cp docker-extension ~/.docker/cli-plugins/
 ```
+
+### Publishing
+
+The extension uses 4 "builder" containers for concurrent builds of the Tanzu CLI,
+the backend utility, the React client, etc.
+
+When building and publishing a new extension image,
+ensure you have [authenticated to the GitHub container registry (ghcr).](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-to-the-container-registry)
+This is where the multi architecture builder images are stored and pushed to.
+When building the extension image for multiple architectures,
+these builder images are used (and pulled from ghcr) to generate the final image
+which is pushed to Dockerhub.
+
+To build everything with a tag, including the extension itself, use:
+
+```sh
+TAG=1.2.3 make build-push-everything
```

Comment:
Maybe this is enough - if `TAG` needs to be specified then that means there has to be some awareness of what they are doing and why.

Reply:
`TAG` doesn't _have_ to be specified. The default tag is `dev` (which was sort of for testing purposes). Maybe we don't set a default tag and just check if it's empty?

Reply:
That could work. Let's see what others think. Maybe I'm being too cautious.